miio/gateway.py

```diff
@@ -202,140 +189,160 @@ def set_developer_key(self, key):
     @command()
     def timezone(self):
         """Get current timezone."""
-        return self.send("get_device_prop", ["lumi.0", "tzone_sec"])
+        return self.get_gateway_prop("tzone_sec")
 
     @command()
     def get_illumination(self):
         """Get illumination. In lux?"""
-        return self.send("get_illumination")[0]
+        return self.send("get_illumination").pop()
 
 
 class GatewayAlarm(Device):
     """Class representing the Xiaomi Gateway Alarm."""
-
-    def __init__(self, parent) -> None:
-        self._device = parent
+    
+    def __init__(
+        self,
+        parent: Gateway = None,
+        ip: str = None,
+        token: str = None,
+        start_id: int = 0,
+        debug: int = 0,
+        lazy_discover: bool = True,
+    ) -> None:
+        if parent is not None:
+            self._gateway = parent
+        else:
+            self._gateway = Device(ip, token, start_id, debug, lazy_discover)
+            _LOGGER.debug("Creating new device instance, only use this for cli interface")
 
     @command(default_output=format_output("[alarm_status]"))
     def status(self) -> str:
         """Return the alarm status from the device."""
         # Response: 'on', 'off', 'oning'
-        return self._device.send("get_arming").pop()
+        return self._gateway.send("get_arming").pop()
 
     @command(default_output=format_output("Turning alarm on"))
     def on(self):
         """Turn alarm on."""
-        return self._device.send("set_arming", ["on"])
+        return self._gateway.send("set_arming", ["on"])
 
     @command(default_output=format_output("Turning alarm off"))
     def off(self):
         """Turn alarm off."""
-        return self._device.send("set_arming", ["off"])
+        return self._gateway.send("set_arming", ["off"])
 
     @command()
     def arming_time(self) -> int:
         """Return time in seconds the alarm stays 'oning' before transitioning to 'on'"""
         # Response: 5, 15, 30, 60
-        return self._device.send("get_arm_wait_time").pop()
+        return self._gateway.send("get_arm_wait_time").pop()
 
     @command(click.argument("seconds"))
     def set_arming_time(self, seconds):
         """Set time the alarm stays at 'oning' before transitioning to 'on'"""
-        return self._device.send("set_arm_wait_time", [seconds])
+        return self._gateway.send("set_arm_wait_time", [seconds])
 
     @command()
     def triggering_time(self) -> int:
         """Return the time in seconds the alarm is going off when triggered"""
         # Response: 30, 60, etc.
-        return self._device.send("get_device_prop", ["lumi.0", "alarm_time_len"]).pop()
+        return self._gateway.get_gateway_prop("alarm_time_len").pop()
```

Comment:
Do you think it would make sense to have all those getters be without gateway? The object itself knows its did, so `self._gateway.get_prop()` and `self.gateway.light.get_prop()` should work just fine.

Reply:
I will remove the gateway

Reply:
done